sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileClientBuilder.java

```diff
@@ -178,6 +178,8 @@ public class ShareFileClientBuilder implements
     private ClientOptions clientOptions = new ClientOptions();
     private Configuration configuration;
     private ShareServiceVersion version;
+    private boolean allowSourceTrailingDot;
```

Comment:
What about the DirectoryClientBuilder and ServiceClientBuilder?

Reply:
There is no DirectoryClientBuilder, ShareDirectoryClient can be built by calling `ShareFileClientBuilder.buildDirectoryClient()`.

In the OneNote, it states "Do not pass these headers on the Service or Share level", so would it make sense to add allowTrailingDot property on the ShareServiceClientBuilder? Also, would that mean we should remove the allowTrailingDot property from the ShareClientBuilder class too?

Reply:
"Do not pass these headers on the Service or Share level" means these parameters shoudn't be passed to the service, this is handled by our generated code.

We want to take these parameters on the Service and Share level, so any clients that are derived from them will inherit these properties.  

For example, in .NET there is a method to get a DirectoryClient from a ShareClient, inheriting all its properties.  I assume similar methods exist in Java?